File: android/app/src/main/kotlin/com/goalmaker/app/domain/planning/PlanningDay.kt
package com.goalmaker.app.domain.planning

import java.time.LocalDate
import java.time.LocalDateTime

/**
 * The day plans belong to: the local date shifted back by the hour the day starts (docs/lists.md),
 * so planning at 01:30 still happens on yesterday's day.
 */
object PlanningDay {
    const val DEFAULT_START_HOUR = 4
    val START_HOURS = 0..6

    fun of(now: LocalDateTime, startHour: Int = DEFAULT_START_HOUR): LocalDate = now.minusHours(startHour.toLong()).toLocalDate()
}

File: windows/src/GoalMaker.Core/Planning/PlanningDay.cs
namespace GoalMaker.Core.Planning;

/// <summary>
/// The day plans belong to: the local date shifted back by the hour the day starts (docs/lists.md),
/// so planning at 01:30 still happens on yesterday's day.
/// </summary>
public static class PlanningDay
{
    public const int DefaultStartHour = 4;
    public const int LatestStartHour = 6;

    public static DateOnly Of(DateTime now, int startHour = DefaultStartHour) => DateOnly.FromDateTime(now.AddHours(-startHour));
}
